File: t/VB/OmrProcessingDemo/UI/IntroDialog.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Partial Public Class IntroDialog
   Inherits Form

   Public Enum StartupType
      None
      NewTemplate
      LoadTemplate
      LoadWorkspace
      LoadBackupTemplate
   End Enum

   Public Property Start As StartupType

   Public Sub New(ByVal recoverVisible As Boolean)
      InitializeComponent()
      Start = StartupType.None
      rdbtnLoadAutosave.Visible = recoverVisible
      lblDescAutoRecover.Visible = recoverVisible
   End Sub

   Private Sub rdbtnCreateNewTemplate_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
      Start = StartupType.NewTemplate
   End Sub

   Private Sub rdbtnLoadTemplate_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
      Start = StartupType.LoadTemplate
   End Sub

   Private Sub rdbtnLoadWorkspace_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
      Start = StartupType.LoadWorkspace
   End Sub

   Private Sub rdbtnLoadAutosave_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
      Start = StartupType.LoadBackupTemplate
   End Sub

   Private Sub IntroDialog_FormClosed(ByVal sender As Object, ByVal e As FormClosedEventArgs)
      If Me.DialogResult = DialogResult.Cancel Then
         Start = StartupType.None
      End If
   End Sub
End Class

File: t/CS/OmrProcessingDemo/UI/IntroDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace OmrProcessingDemo.UI
{
   public partial class IntroDialog : Form
   {
      public enum StartupType
      {
         None,
         NewTemplate,
         LoadTemplate,
         LoadWorkspace,
         LoadBackupTemplate
      }

      public StartupType Start { get; private set; }

      public IntroDialog(bool recoverVisible)
      {
         InitializeComponent();

         Start = StartupType.None;

         rdbtnLoadAutosave.Visible = recoverVisible;
         lblDescAutoRecover.Visible = recoverVisible;
      }

      private void rdbtnCreateNewTemplate_CheckedChanged(object sender, EventArgs e)
      {
         Start = StartupType.NewTemplate;
      }

      private void rdbtnLoadTemplate_CheckedChanged(object sender, EventArgs e)
      {
         Start = StartupType.LoadTemplate;
      }

      private void rdbtnLoadWorkspace_CheckedChanged(object sender, EventArgs e)
      {
         Start = StartupType.LoadWorkspace;
      }
      private void rdbtnLoadAutosave_CheckedChanged(object sender, EventArgs e)
      {
         Start = StartupType.LoadBackupTemplate;
      }

      private void IntroDialog_FormClosed(object sender, FormClosedEventArgs e)
      {
         if (this.DialogResult == DialogResult.Cancel)
         {
            Start = StartupType.None;
         }
      }

   }
}
